pkg/cmd/operator/sidecarprobe.go

```diff
@@ -0,0 +1,173 @@
+	o := NewSidecarProbeOptions(streams)
+
+	cmd := &cobra.Command{
+		Use:   "sidecar-probe",
+		Short: "Run the scylla sidecar probe.",
+		Long:  `Run the scylla sidecar probe.`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			err := o.Validate()
+			if err != nil {
+				return err
+			}
+
+			err = o.Complete()
+			if err != nil {
+				return err
+			}
+
+			err = o.Run(streams, cmd)
+			if err != nil {
+				return err
+			}
+
+			return nil
+		},
+
+		SilenceErrors: true,
+		SilenceUsage:  true,
+	}
+
+	cmd.Flags().StringVarP(&o.ListenAddress, "listen-address", "", o.ListenAddress, "Address on which probes server listen.")
+
+	return cmd
+}
+
+func (o *SidecarProbeOptions) Validate() error {
+	var errs []error
+
+	if len(o.ListenAddress) == 0 {
+		errs = append(errs, fmt.Errorf("listen-address cannot be empty"))
+	}
+
+	return utilerrors.NewAggregate(errs)
+}
+
+func (o *SidecarProbeOptions) Complete() error {
+	return nil
+}
+
+func (o *SidecarProbeOptions) Run(streams genericclioptions.IOStreams, cmd *cobra.Command) error {
+	klog.Infof("%s version %s", cmd.Name(), version.Get())
+	cliflag.PrintFlags(cmd.Flags())
+
+	stopCh := signals.StopChannel()
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	go func() {
+		<-stopCh
+		cancel()
+	}()
+
+	mux := http.NewServeMux()
+	mux.HandleFunc(naming.LivenessProbePath, handleHealthz)
+	mux.HandleFunc(naming.ReadinessProbePath, handleReadyz)
+
+	server := &http.Server{
+		Addr:    o.ListenAddress,
+		Handler: mux,
+	}
+
+	var wg sync.WaitGroup
+	defer wg.Wait()
+
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+
+		klog.InfoS("Starting probe server")
+		defer klog.InfoS("Probe server shut down")
+
+		err := server.ListenAndServe()
+		if err != nil && !errors.Is(err, http.ErrServerClosed) {
+			klog.Fatalf("ListenAndServe failed: %v", err)
+		}
+	}()
+
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+
+		<-ctx.Done()
+
+		klog.InfoS("Shutting down probe server")
+		defer klog.InfoS("Shut down probe server")
+
+		shutdownCtx, shutdownCtxCancel := context.WithTimeout(context.Background(), 15*time.Second)
+		defer shutdownCtxCancel()
+		err := server.Shutdown(shutdownCtx)
+		if err != nil {
+			klog.ErrorS(err, "Shutting down probe server")
+		}
+	}()
+
+	<-ctx.Done()
+
+	return nil
+}
+
+const (
+	localhost    = "localhost"
+	probeTimeout = 60 * time.Second
+)
+
+func handleHealthz(w http.ResponseWriter, req *http.Request) {
+	w.WriteHeader(http.StatusOK)
+}
+
+func handleReadyz(w http.ResponseWriter, req *http.Request) {
+	ctx, ctxCancel := context.WithTimeout(req.Context(), probeTimeout)
+	defer ctxCancel()
+
+	scyllaClient, err := controllerhelpers.NewScyllaClientForLocalhost()
+	if err != nil {
+		klog.ErrorS(err, "healthz probe: can't get scylla client")
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	defer scyllaClient.Close()
+
+	gossiperRunning, err := scyllaClient.GossiperIsRunning(ctx, localhost)
```

Comment:
shall we verify with someone from the cluster team whether `/storage_service/gossiping` is the right endpoint to check here? Or have you done that already?

Reply:
I asked and apparently there's none telling that port used for inter-node communication is open. 
This one returns true after port is open but it's undocumented and untested for this particular case. So we cannot rely on it that it will behave like this in future versions.
I also tried implementing TCP probe by dialing to this port, but unfortunately ScyllaDB prints and error about connection being closed which would make our users confused that something is wrong.

I filed https://github.com/scylladb/scylladb/issues/16939 which we could shift to once it's available in all supported ScyllaDB versions. 
Until it's missing, I believe it's the best alternative.

Reply:
> Until it's missing, I believe it's the best alternative.

If core says it's undocumented and untested, and there are no other viable alternatives, then I think we shouldn't use it and freeze the PR until we are able to implement in correctly. First of all, this issue is a good leverage to actually get the endpoint necessary to implement this in a reasonable time. If we use a workaround, then not only will we most probably get stuck with it, but also when it breaks, it will be on us for using an undocumented and untested approach. We should stop acting like a third-party.

Reply:
I don't think we should hold important bug fix for such simple thing. Gossiping endpoint is reliable in my view, because it returns whether nodes exchange Gossip. How can you exchange something without listening? 

Keep in mind, that what we ask for is a new feature. New features are not backported to older versions. Do you want to hold it for years Enterprise version having support for it is released?
We already asked for one endpoint and we still didn't get it after 3 years.

Reply:
Resolved in internal conversations